[PR] Personality Traits Checklist Selection & Visualization

harismah opened the pull request:
### Description

This PR implements the **Personality Traits Selection and Visualization** user story. It introduces a front-end interface allowing users to visually select and deselect personality traits in their edit profile page and updates these selections to the backend via API.

### Related User Story

- **[User Story #171: Personality Traits Selection and Visualization](https://github.com/CodePioneerX/Dwella/issues/171)**

### Related Tasks

- **[Task #172: Create front-end interface for displaying and selecting personality traits](https://github.com/CodePioneerX/Dwella/issues/172)**
- **[Task #173: Integrate with API to save updated personality traits](https://github.com/CodePioneerX/Dwella/issues/173)**

### Changes Implemented

- Developed a component that displays personality traits with toggleable icons.
- Integrated API call to save updated personality traits to the backend.
- Updated the profile edit interface to reflect selected, deselected, and default states.

harismah committed:
Update EditProfile.jsx for Personality Checklist

- Display list of personality traits with selectable icons
- Default icon for unselected traits, green check for selected, and red X for removed
- Integrate with backend to save updated traits upon profile submission

harismah committed:
Update userRoutes.js for Personality Trait Checklist

- Modified `update-profile` endpoint to store personalityTraits array in the database
- Enhanced `fetch-profile-edit` endpoint to retrieve and include personalityTraits

Commit:
Prettified Code!

harismah commented:
These are the different icons a user should be able to see:
![image](https://github.com/user-attachments/assets/db01a6bd-5433-4940-92dc-843b9c41d5b7)

harismah committed:
Update userRoutes.js

Removed additional line, to satsify superlinter

Commit:
Prettified Code!

MyWaterBottle reviewed (changes requested):
Personality Traits feature works!

However, I am unable to login even if the user is verified
![image](https://github.com/user-attachments/assets/d3f3393a-cb41-452d-93af-1db48869e5d3)

harismah committed:
Update userRoutes.js

Commit:
Prettified Code!

harismah committed:
Update userRoutes.js

Condition of user being verified was missing from my local branch that was present in the main branch

Commit:
Prettified Code!

MyWaterBottle commented:
Problem has been fix and Personality Traits feature works well! The database reflects the proper changes
